lib/relations/Relation.js

```diff
@@ -74,7 +74,7 @@ class Relation {
     if (this.joinTableModelClass) {
       const knex = this.ownerModelClass.knex();
       if (knex !== this.joinTableModelClass.knex()) {
-        return this.joinTableModelClass.bindKnex(knex);
+        this.joinTableModelClass.knex(knex);
```

Comment:
@koskimas this works indeed! All tests should still pass.

Reply:
This is a huge no-no! Think about parallel transactions and what will happen with them.

Reply:
Oh ok, yeah I hadn't thought about that! 😬 I'll revert the commit then, makes a lot of sense. Surprised that no unit test caught it?